src/WebJobs.Script/Binding/ExtensionBinding.cs

```diff
@@ -67,13 +70,32 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
                 var result = await context.Binder.BindAsync<JObject>(_attributes.ToArray());
                 if (Access == FileAccess.Read)
                 {
-                    context.Value = result;
+                    inputValue = result;
                 }
             }
+            else if (_binding.DefaultType == typeof(JArray))
+            {
+                JArray entityArray = await context.Binder.BindAsync<JArray>(_attributes.ToArray());
+                inputValue = entityArray;
+            }
             else
             {
                 throw new NotSupportedException($"ScriptBinding type {_binding.DefaultType} is not supported");
             }
+
+            if (inputValue != null)
+            {
+                if (context.DataType == DataType.Stream)
```

Comment:
Just do what you need to do to unblock then, and I'll revisit soon. I was just pointing out that this method is now a little hard to understand (not your fault) and I want to investigate improvements.